### PR TITLE
chore(docs): v4 migration page (VIV-1801)

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

File: docs/resources/v4-release-migration.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# vivid-4.0.0
+
+V-4 is currently scheduled for the 1st July 2024.  
+Vivid v-4 is a small iteration, not a rewrite like the last major release.  
+This is a migration guide to make the upgrade easy :)
```
Comment (TaylorJ76) [inline docs/resources/v4-release-migration.md:5]:
```suggestion
# Vivid v4.0.0 Migration Guide

Release date: 1st July 2024.  
This document provides descriptions of the breaking changes in this release and instructions on how to prepare for them.
```

File: docs/resources/v4-release-migration.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+---
+
+## Audio-player
+
+### The change
```
Comment (TaylorJ76) [inline docs/resources/v4-release-migration.md:11]: Could we provide a short description of the change for each? 
For example: 

The audio-player will now stretch to fill it's container's width by default. Previously it was restricted to 350px.

File: docs/resources/v4-release-migration.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+Form elements will now forward their value props as DOM properties instead of attributes.
+
+|                     | V-3                                                                       | V-4                                                            |
+| ------------------- | ------------------------------------------------------------------------- | -------------------------------------------------------------- |
+| Form elements props | the value prop holds the initial value and currentValue the current value | value holds the current value and initialValue the initial one |
+
+### How to get ready?
+
+TBA: Look into how we help with an upgrade script?
```
Comment (TaylorJ76) [inline docs/resources/v4-release-migration.md:100]: Maybe mention something about the eslint-plugin? Ask Richard for some help with the text.

**Review by TaylorJ76 — CHANGES_REQUESTED** [review]

File: docs/resources/v4-release-migration.md (hunk 3 above)
Comment (rachelbt) [reply docs/resources/v4-release-migration.md:100]: I did, he said that he will send me a line or two for this section :)

File: docs/resources/v4-release-migration.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+|                                                                   | V-3                                | V-4                    |
+| ----------------------------------------------------------------- | ---------------------------------- | ---------------------- |
+| `currentValue` / `currentChecked` / `currentStart` / `currentEnd` | Controls current value             | Removed                |
+| `value`/`checked`                                                 | Inconsistent between Vue 2 / Vue 3 | Controls current value |
+| `initialValue` / `defaultChecked`                                 | Doesn't exist                      | Controls initial value |
+
+## How to get ready?
+
+- First, make sure you're on the latest version of Vivid 2.x.
```
Comment (TaylorJ76) [inline docs/resources/v4-release-migration.md:147]: Should this say Vivid 2.x?

**Review by TaylorJ76 — CHANGES_REQUESTED** [review]

File: docs/resources/v4-release-migration.md (hunk 4 above)
Comment (rachelbt) [reply docs/resources/v4-release-migration.md:147]: suppose to be 3.x :)

**Review by TaylorJ76 — DISMISSED** [review]

File: docs/resources/v4-release-migration.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+- VRangeSlider
+
+Usage of these props is discouraged. To set the current value, you should use the `v-model` instead:
+
+```html
+<VTextField v-model="value" />
+<!-- Which is syntactic sugar for: -->
+<VTextField
+	:model-value="value"
+	@update:model-value="$event => (value = $event)"
```
Comment (RichardHelm) [inline docs/resources/v4-release-migration.md:131]: Actually needs to be this:
```suggestion
	@update:modelValue="$event => (value = $event)"
```

File: docs/resources/v4-release-migration.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+<vwc-note connotation="information" icon="info-solid" headline="This is not available yet">
+It will be introduced in Version 4
+</vwc-note>
+
+- First, make sure you're on the latest version of Vivid 3.x.  
+  Optionally, install the latest version of our new ESLint plugin, which can help with the migration.
+
+- Replace all usage of `currentValue` / `currentChecked` with `modelValue`.
+- Replace `currentStart` / `currentEnd` with `start` / `end`.  
+  The `no-current-value-attribute` rule of our ESLint plugin can perform this migration automatically.
```
Comment (RichardHelm) [inline docs/resources/v4-release-migration.md:156]: Importantly, the rule applies to both points:
```suggestion
- Replace all usage of `currentValue` / `currentChecked` with `modelValue` and all usages `currentStart` / `currentEnd` with `start` / `end`.  
  The `no-current-value-attribute` rule of our ESLint plugin can perform this migration automatically.
```

File: docs/resources/v4-release-migration.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+| ----------------------------------------------------------------- | ---------------------------------- | ---------------------- |
+| `currentValue` / `currentChecked` / `currentStart` / `currentEnd` | Controls current value             | Removed                |
+| `value`/`checked`                                                 | Inconsistent between Vue 2 / Vue 3 | Controls current value |
+| `initialValue` / `defaultChecked`                                 | Doesn't exist                      | Controls initial value |
+
+## How to get ready?
+
+<vwc-note connotation="information" icon="info-solid" headline="This is not available yet">
+It will be introduced in Version 4
+</vwc-note>
```
Comment (RichardHelm) [inline docs/resources/v4-release-migration.md:149]: Not true, it will go in 3.x
```suggestion
```

File: docs/resources/v4-release-migration.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+### How to get ready?
+
+set `max-inline-size` or `inline-size` directly on audio-player or on its parent.
+
+---
+
+## Data-grid
+
+Data-grid-cell will no longer have a fixed block-size but will adjust to its content and its text will not be trimmed by default ad before.
```
Comment (RichardHelm) [inline docs/resources/v4-release-migration.md:27]: This is only for non-header cells:
```suggestion
Data-grid-cells outside of the table header will no longer have a fixed block-size but will adjust to its content and its text will not be trimmed by default as before.
```

**Review by RichardHelm — CHANGES_REQUESTED** [review]

**Review by RichardHelm — APPROVED** [review]

**Pull request closed without merging**